src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotResumeTest.java

```diff
@@ -89,51 +88,23 @@ public void verifySnapshotIsResumedFromKey() throws Exception {
 		// Only verifying the record count since the snapshot records are not ordered, so it may be
 		// a little complex to verify them in the sorted order at the moment
 		final int finalTotalConsumedSoFar = totalConsumedSoFar;
-		CompletableFuture.runAsync(() -> verifyRecordCount(recordsCount - finalTotalConsumedSoFar))
-			.exceptionally(throwable -> {
-				throw new RuntimeException(throwable);
-			}).get();
-	}
-
-	private void verifyRecordCount(long recordsCount) {
-		waitAndFailIfCannotConsumeRecords(new ArrayList<>(), recordsCount);
-	}
+		List<SourceRecord> records = new ArrayList<>();
+		waitAndFailIfCannotConsume(records, recordsCount - finalTotalConsumedSoFar);
 
-	private void waitAndFailIfCannotConsumeRecords(List<SourceRecord> records, long recordsCount) {
-		waitAndFailIfCannotConsumeRecords(records, recordsCount, 300 * 1000 /* 5 minutes */);
-	}
+		// Consume the remaining records as there can be duplicates records while resuming snapshot,
+		// they should NOT be equal to what we have consumed before connector restart.
+		LOGGER.info("Consuming remaining records (if available)");
 
-	/**
-	 * Consume the records available and add them to a list for further assertion purposes.
-	 * @param records list to which we need to add the records we consume, pass a
-	 * {@code new ArrayList<>()} if you do not need assertions on the consumed values
-	 * @param recordsCount total number of records which should be consumed
-	 * @param milliSecondsToWait duration in milliseconds to wait for while consuming
-	 */
-	private void waitAndFailIfCannotConsumeRecords(List<SourceRecord> records, long recordsCount,
-																								 long milliSecondsToWait) {
-		AtomicLong totalConsumedRecords = new AtomicLong();
-		long seconds = milliSecondsToWait / 1000;
-		try {
-			Awaitility.await()
-				.atMost(Duration.ofSeconds(seconds))
-				.until(() -> {
-					int consumed = consumeAvailableRecords(record -> {
-						LOGGER.debug("The record being consumed is " + record);
-						records.add(record);
-					});
-					if (consumed > 0) {
-						totalConsumedRecords.addAndGet(consumed);
-						LOGGER.debug("Consumed " + totalConsumedRecords + " records");
-					}
-
-					return totalConsumedRecords.get() == recordsCount;
-				});
-		} catch (ConditionTimeoutException exception) {
-			fail("Failed to consume " + recordsCount + " in " + seconds + " seconds", exception);
+		int remainingConsumedRecords = 0;
+		for (int i = 0; i < 10; ++i) {
+			remainingConsumedRecords += consumeAvailableRecords(record -> {});
+			TestHelper.waitFor(Duration.ofSeconds(2));
 		}
 
-		assertEquals(recordsCount, totalConsumedRecords.get());
+		LOGGER.info("Remaining consumed record count: {}", remainingConsumedRecords);
+		assertNotEquals(finalTotalConsumedSoFar, remainingConsumedRecords);
+
+		assertEquals(recordsCount - finalTotalConsumedSoFar, records.size());
```

Comment:
To confirm that explicitCheckpointing works we need to assert that the "remaining consumed records" is equal to an appropriate value (100 ? size of.2 batches)